src/engine/Server.cpp

```diff
@@ -830,9 +838,13 @@ net::awaitable<std::optional<Server::PlannedQuery>> Server::parseAndPlan(
   // probably related to issues in GCC's coroutine implementation.
   return computeInNewThread(
       [&query, &qec, enablePatternTrick = enablePatternTrick_,
-       handle = std::move(handle),
-       timeLimit]() mutable -> std::optional<PlannedQuery> {
+       handle = std::move(handle), timeLimit,
+       maxSend]() mutable -> std::optional<PlannedQuery> {
         auto pq = SparqlParser::parseQuery(query);
+        if (maxSend.has_value()) {
+          pq._limitOffset._limit =
+              std::min(maxSend.value(), pq._limitOffset.limitOrDefault());
+        }
```

Comment:
Applying this to the parsed query after parsing is the correct move in my opinion, but has the side-effect that basically every query being executed within qlever-ui now has a visible limit in the analysis tree. Previously it was only implicitly limited, taking up more space in cache.

Reply:
On current master the limit code on operations is effectively dead code. So there are 3 options on how to proceed from there:
1. Throw out the exporter code, correct the query planner and have operations be "aware" of their offset/limit clauses. The exporter can then safely assume the result is already the correct shape (What this PR is doing)
2. Keep both approaches and either write code that is mostly (~50%-70%) duplicated and unused or put an AD_FAIL assertion there because it will never be called. Really bad for maintainability. Also in my opinion this behaviour is a bug in the current implementation that didn't fail so far because there are no operations that return true for `supportsLimit()` and handle the case for offsets != 0 properly themselves.
3. Keep the Code in the exporter and drop the limit clause within operations. In this case it's the exporters call to only consume as much as needed. If the `IdTable` is fully materialized and not some sort of generator, then the exporter would have to crop the results accordingly. In this case if any operations need only part of their child operations they would have to crop stuff on their own.

So in essence the problem with the limit stuff is that I either have to write code to support the framework it's designed around.
Option 2 (the status quo) is a dealbreaker in my opinion, I don't see the point of writing code that will never get called or write code that simply throws or does nothing to avoid having to write code that will never get called.
Option 1 is a rather simple solution to this problem and has the nice benefit of "fixing" the issue that the "send" parameter is now also obeyed for csv/tsv just like that. (As you can see the diff is rather small).
Option 3 is also a reasonable approach in my opinion, but requires slightly more work than this (probably)